indexing-service/src/test/java/org/apache/druid/indexing/common/task/CompactionTaskTest.java

```diff
@@ -1492,7 +1492,7 @@ public void testQueryGranularityAndNullSegmentGranularity() throws IOException
         null,
         null,
         null,
-        new ClientCompactionTaskGranularitySpec(null, new PeriodGranularity(Period.months(3), null, null), null),
+        new ClientCompactionTaskGranularitySpec(null, Granularities.HOUR, null),
```

Comment:
Related to [this](https://github.com/apache/druid/pull/16324/files#r1577256445) comment, please keep the test inputs as they are and only update the expected result. In this case, the test should throw an exception due to the added validation. Separately, we can add a new test with valid input as you have done here.

Reply:
Done